plugins/modules/fabric_devices_workflow_manager.py

```diff
@@ -0,0 +1,4080 @@
+            with the Border Settings, L2 Handoff, L3 SDA Handoff, L3 IP Handoff information.
+        Returns:
+            self: The current object with updated desired fabric devices information.
+        Description:
+            Do all the validation which is done in the GUI and format the payload
+            in such a way that we can pass it to the API.
+            Call the respective functions for gathering the device details,
+            L2 Handoff, L3 IP Handoff, L3 SDA Handoff provided by the user.
+        """
+
+        fabric_devices_details = []
+        fabric_name = fabric_devices.get("fabric_name")
+        device_config = fabric_devices.get("device_config")
+        device_config_index = -1
+        for item in device_config:
+            device_config_index += 1
+            fabric_site_id = self.have.get("fabric_devices")[device_config_index].get("fabric_site_id")
+            device_ip = fabric_devices.get("device_ip")
+            fabric_devices_info = {
+                "device_details": None,
+                "l2_handoff_details": [],
+                "sda_l3_handoff_details": None,
+                "ip_l3_handoff_details": [],
+            }
+            network_device_id = self.have.get("fabric_devices")[device_config_index].get("network_device_id")
+            fabric_devices_info.update({
+                "device_details": self.get_device_params(fabric_site_id, network_device_id, item, device_config_index),
+                "l2_handoff_details": self.get_l2_handoff_params(fabric_site_id, network_device_id, item, device_config_index),
+                "sda_l3_handoff_details": self.get_sda_l3_handoff_params(fabric_site_id, network_device_id, item, device_config_index),
+                "ip_l3_handoff_details": self.get_ip_l3_handoff_params(fabric_site_id, network_device_id, item,
+                                                                       device_config_index, fabric_name),
+            })
+            self.log(
+                "The fabric device with IP '{ip}' details under the site '{site}': {details}"
+                .format(ip=device_ip, site=fabric_name, details=fabric_devices_info)
+            )
+            fabric_devices_details.append(fabric_devices_info)
+
+        self.log("SDA fabric devices playbook details: {requested_state}"
+                 .format(requested_state=fabric_devices_details), "DEBUG")
+        self.want.update({"fabric_devices": fabric_devices_details})
+        self.msg = "Collecting the SDA fabric transits details from the playbook."
+        self.status = "success"
+        return self
+
+    def get_want(self, config):
+        """
+        Get the SDA fabric devices related information from playbook.
+
+        Parameters:
+            config (list of dict): Playbook details contains the details of fabric devices.
+        Returns:
+            self: The current object with updated fabric devices details.
+        Description:
+
+        """
+
```

Comment:
self.log("Starting to retrieve fabric devices information from the provided configuration.", "DEBUG")